plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -146,3 +152,72 @@ def test(self):
                 data=str(error),
             )
         return {"success": True}
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Carbon Black Cloud")
+
+        alerts_endpoint = self.base_url + f"/api/alerts/v7/orgs/{self.org_key}/alerts/_search"
+        observations_endpoint = self.base_url + f"/api/investigate/v2/orgs/{self.org_key}/observations/search_jobs"
+
+        current_time = datetime.now(timezone.utc)
+        now = current_time.strftime(TIME_FORMAT)
+        minus_five_mins = (current_time - timedelta(minutes=5)).strftime(TIME_FORMAT)
+
+        fd = {
+            alerts_endpoint: {
+                "time_range": {"start": minus_five_mins, "end": now},
+                "criteria": {},
+                "start": "1",
+                "rows": str(1),  # max number of results that can be returned
+                "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+            },
+            observations_endpoint: {
+                "rows": 1,
+                "start": 0,
+                "fields": ["*"],
+                "criteria": {"observation_type": OBSERVATION_TYPES},
+                "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+                "time_range": {"start": minus_five_mins, "end": now},
+            },
+        }
+
+        return_msg = "Task connection test to Carbon Black Cloud successful"
+        return_msg_error = "Task connection test to Carbon Black Cloud failed"
+        self.logger.info("Testing get alerts")
```

Comment:
Also minor, this could be passed into the for loop and log the key if it gets changed to be `alerts` and `observations` and then we could remove the comment on L193 as it's not just the search jobs endpoint?

Reply:
good spot didn't notice this